Submit SunPy to the Journal of Open Source Software (#3391)
sunpy/sunpy: joss_paper/paper.md

---
title: 'SunPy: A Python package for Solar Physics'
tags:
  - Python
  - Astronomy
  - Solar physics
authors:
  - name: Stuart J. Mumford
    orcid: 0000-0003-4217-4642
    affiliation: "1, 2"

>   - name: Nabil Freij

Reviewer: ```suggestion
  - name: Nabil Freij
    orchid: 0000-0002-6253-082X
    affiliation: "Institute for Environmental Analytics, University of Reading, Reading RG6 6BX"
```

>   - name: Steven Christe

Reviewer 2: ```suggestion
  - name: Steven Christe
    orcid: 0000-0001-6127-795X
    affiliation: "NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA"
```

  - name: Jack Ireland
  - name: Florian Mayer
  - name: Keith Hughitt

>   - name: Albert Y. Shih

Reviewer 3: ```suggestion
  - name: Albert Y. Shih
    orcid: 0000-0001-6874-2594
    affiliation: "NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA"
```

  - name: Daniel Ryan
  - name: Simon Liedtke

>   - name: David Pérez-Suárez

Reviewer 4: ```suggestion
  - name: David Pérez-Suárez
    orcid: 0000-0003-0784-6909
    affiliation: "University College London, Gower Street, London, UK"
```

  - name: Pritish Chakraborty
  - name: Vishnunarayan K I.

>   - name: Andrew Inglis

Reviewer 5: ```suggestion
  - name: Andrew Inglis
   orchid: 0000-0003-0656-2437
   affiliation: "Catholic University of America / NASA Goddard Space Flight Center, Greenbelt, MD, 20771, USA"
```

  - name: Punyaslok Pattnaik

>   - name: Brigitta Sipocz

Reviewer 6: ```suggestion
  - name: Brigitta Sipőcz
    orchid: https://orcid.org/0000-0002-3713-6337
    affiliation: "DIRAC Institute, Department of Astronomy, University of Washington, Seattle, WA 98195, USA"
```

Author: git shortlog did not enjoy the correct spelling of your surname clearly!

Reviewer 6: yes, I'm old school with e-mail and still do ascii, only even when writing Hungarian.

  - name: Rishabh Sharma

>   - name: Andrew Leonard

Reviewer 7: ```suggestion
  - name: Andrew Leonard
    orcid: 0000-0001-5270-7487
    affiliation: Aperio Software Ltd., Headingley Enterprise and Arts Centre, Bennett Rd, Leeds LS6 3HN
```

>   - name: David Stansby

Reviewer 8: ```suggestion
  - name: David Stansby
     orcid: 0000-0002-1365-1908
     affiliation: 'Mullard Space Science Laboratory, University College London, Surrey, UK'
```

  - name: Russell Hewett
  - name: Alex Hamilton

>   - name: Laura Hayes

Reviewer 9: -name: Laura Hayes
 orcid: 0000-0002-6835-2390
 affiliation: 'NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA'

  - name: Asish Panda
  - name: Matt Earnshaw

>   - name: Nitin Choudhary

Reviewer 10: orcid : 0000-0001-6915-4583
affiliation: 'Department of Mathematics, Indian Institute of Technology, Kharagpur, 721302, India'

  - name: Ankit Kumar

>   - name: Prateek Chanda

Reviewer 11: ```suggestion
  - name: Prateek Chanda
    orcid: 0000-0002-7068-2866
    affiliation: 'Department of Computer Science & Technology, Indian Institute of Engineering Science & Technology, Shibpur, 711103, India'
```

>   - name: Md Akramul Haque

Reviewer 12: ```suggestion
  - name: Md Akramul Haque
    affiliation: 'Department of Mechanical Engineering, ZHCET, Aligarh Muslim University, Aligarh, India'
```

>   - name: Michael S Kirk

Reviewer 13: ```suggestion
  - name: Michael S Kirk
    orcid: 0000-0001-9874-1429
    affiliation: 'Catholic University of America / NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA'
```

  - name: Michael Mueller
  - name: Sudarshan Konge
  - name: Rajul Srivastava
  - name: Yash Jain
  - name: Samuel Bennett
  - name: Ankit Baruah

>   - name: Will Barnes

Reviewer 14: ```suggestion
  - name: Will Barnes
     affiliation: Lockheed Martin Solar and Astrophysics Laboratory / Bay Area Environmental Research Institute
     orcid: 0000-0001-9642-6089
```

  - name: Michael Charlton

>   - name: Cubostar

Reviewer 15: ```suggestion
  - name: Cubostar
```

>   - name: Shane Maloney

Reviewer 15: ```suggestion
  - name: Shane Maloney
```

>   - name: Shane Maloney

Reviewer 16: ```suggestion
  - name: Shane Maloney
     orcid: 0000-0002-4715-1805
     affiliation: Trinity College Dublin / Dublin Institute for Advanced Studies
```

>   - name: Nicky Chorley

Reviewer 17: ```suggestion
  - name: Nicky Chorley
  - affiliation: Centre for Fusion, Space and Astrophysics, Physics Department, University of Warwick, Coventry CV4 7AL, United Kingdom
```

>   - name: Nicky Chorley

Reviewer 17: ```suggestion
  - name: Nicky Chorley
  - orcid: 0000-0002-2747-2716
```

Decided to register for an ORCID after I added my affiliation, so including it now!
Finally registered for an ORCID, so adding

Author: I added your orcid to the other comment.

  - name: Himanshu
  - name: Sanskar Modi
  - name: James Paul Mason
  - name: Naman9639
  - name: Yash

>   - name: Jose Ivan Campos Rozo

Reviewer 18: Name: Jose Ivan Campos Rozo
orcid: 0000-0001-8883-6790
affiliation: Institut für Physik/IGAM - Karl-Franzens University of Graz, Austria

  - name: Larry Manley
  - name: Agneet Chatterjee
  - name: John Evans
  - name: Michael Malocha

>   - name: Monica Bobra

Reviewer 19: ```suggestion
  - name: Monica G. Bobra
  - affiliation: "W.W. Hansen Experimental Physics Laboratory, Stanford University, Stanford, CA 94305, USA"
  - orcid: 0000-0002-5662-9604
```

  - name: Sourav Ghosh
  - name: Airmansmith97

>   - name: Dominik Stańczak

Reviewer 20: ```suggestion
  - name: Dominik Stańczak
  - affiliation: University of Warsaw
  - orcid: 0000-0001-6291-8843
```

  - name: Ruben De Visscher

>   - name: Shresth Verma

Reviewer 21: ```suggestion
  - name: Shresth Verma
  - affiliation: ABV-Indian Institute of Information Technology and Management, Gwalior
  - orcid:  0000-0003-0370-5471
```

  - name: Ankit Agrawal
  - name: Dumindu Buddhika
  - name: Swapnil Sharma
  - name: Jongyeob Park
  - name: Matt Bates
  - name: Dhruv Goel
  - name: Garrison Taylor
  - name: Goran Cetusic
  - name: Jacob
  - name: Mateo Inchaurrandieta
  - name: Sally Dacie
  - name: Sanjeev Dubey
  - name: Deepankar Sharma
  - name: Erik M. Bray

>   - name: Jai Ram Rideout

Reviewer 22: ```suggestion
  - name: Jai Ram Rideout
    orcid: 0000-0003-2587-1454
    affiliation: "Dogfox Software LLC, Flagstaff, AZ 86001, USA"
```

  - name: Serge Zahniy
  - name: Tomas Meszaros
  - name: Abhigyan Bose
  - name: Andre Chicrala
  - name: Ankit
  - name: Chloé Guennou
  - name: Daniel D'Avella

>   - name: Daniel Williams

Reviewer 23: ```suggestion
  - name: Daniel Williams
  - affiliation: "SUPA, University of Glasgow, Glasgow G12 8QQ, United Kingdom"
  - orcid: 0000-0003-3772-198X
```

  - name: Jordan Ballew
  - name: Nick Murphy
  - name: Priyank Lodha
  - name: Thomas Robitaille
  - name: Yash Krishan
  - name: Andrew Hill

>   - name: Arthur Eigenbrot

Reviewer 24: ```suggestion
  - name: Arthur Eigenbrot
    orcid: 0000-0003-0810-4368
    affiliation: "National Solar Observatory" 
```

  - name: Benjamin Mampaey
  - name: Bernhard M. Wiedemann

>   - name: Carlos Molina

Reviewer 25: - name: Carlos Molina
    orcid: 0000-0003-0300-4106

>   - name: Carlos Molina

Reviewer 25: - name: Carlos Molina
orcid: 0000-0003-0300-4106

  - name: Duygu Keşkek
  - name: Ishtyaq Habib

>   - name: Joe Letts

Reviewer 26: ```suggestion
  - name: Joseph Letts
  orcid: 0000-0001-9900-739X
```

>   - name: Juanjo Bazán

Reviewer 27: ```suggestion
  - name: Juanjo Bazán
    orcid: 0000-0001-7699-3983
    affiliation: CIEMAT, Astroparticle physics, Madrid, Spain.
```

>   - name: Quinn Arbolante

Reviewer 15: ```suggestion
  - name: Quinn Arbolante
     affiliation: Lockheed Martin Solar and Astrophysics Laboratory
     orcid: 0000-0003-0260-453X
```

  - name: Reid Gomillion
  - name: Yash Kothari
  - name: Yash Sharma

>   - name: Abigail Stevens

Reviewer 28: orcid: 0000-0002-5041-3079
affiliation: 'Department of Physics & Astronomy, Michigan State University, 567 Wilson Road, East Lansing, MI 48824, USA'
affiliation: 'Department of Astronomy, University of Michigan, 1085 South University Avenue, Ann Arbor, MI 48109, USA'

>   - name: Abigail Stevens

Reviewer 28: ```suggestion:
name: Abigail L. Stevens
orcid: 0000-0002-5041-3079
affiliation: 'Department of Physics & Astronomy, Michigan State University, 567 Wilson Road, East Lansing, MI 48824, USA'
affiliation: 'Department of Astronomy, University of Michigan, 1085 South University Avenue, Ann Arbor, MI 48109, USA'
```

>   - name: Adrian Price-Whelan

Reviewer 29: ```suggestion
  -name: Adrian Price-Whelan
    orcid: 0000-0003-0872-7098
    affiliation: Center for Computational Astrophysics, Flatiron Institute, 162 Fifth Ave, New York, NY 10010, USA
```

  - name: Ambar Mehrotra
  - name: Arseniy Kustov
  - name: Brandon Stone

>   - name: Dang Trung Kien

Reviewer 30: ```suggestion
  - name: Trung Kien Dang
    orcid: 0000-0001-7562-6495
    affiliation: Saw Swee Hock School of Public Health, National University Health System, National University of Singapore, Singapore
```

  - name: Emmanuel Arias

>   - name: Fionnlagh Mackenzie Dover

Reviewer 31: ```suggestion
  - name: Fionnlagh Mackenzie Dover
 orcid: 0000-0002-1984-7303
 affiliation: SP2RC, School of Mathematics and Statistics, The University of Sheffield, UK
```

>   - name: Freek Verstringe

Reviewer 32: ```suggestion
  - name: Freek Verstringe
    affiliation: "Royal Observatory of Belgium"
```

>   - name: Gulshan Mittal

Reviewer 33: name: Gulshan Kumar
orcid: 0000-0001-8523-7223
affiliation: International Institute of Information Technology, Hyderabad

>   - name: Gulshan Mittal

Reviewer 33: ```suggestion
  name: Gulshan Kumar
  orcid: 0000-0001-8523-7223
  affiliation: International Institute of Information Technology, Hyderabad
```

  - name: Harsh Mathur
  - name: Igor Babuschkin
  - name: Jaylen Wimbish
  - name: Juan Camilo Buitrago-Casas
  - name: Kalpesh Krishna

>   - name: Kaustubh Hiware

Reviewer 34: ```suggestion
  - name: Kaustubh Hiware
    orcid: 0000-0003-3301-1016
    affiliation: "Indian Institute of Technology, Kharagpur"
```

  - name: Manas Mangaonkar
  - name: Matthew Mendero
  - name: Mickaël Schoentgen
  - name: Norbert Gyenge

>   - name: Ole Streicher

Reviewer 35: orcid: 0000-0001-7751-1843
affiliation: Leibniz Institute for Astrophysics Potsdam, Germany

  - name: Rajasekhar Reddy Mekala
  - name: Rishabh Mishra

>   - name: S Shashank

Reviewer 36: ```suggestion
  - name: Shashank Srikanth
    affiliation: International Institute of Information Technology, Hyderabad
```

  - name: Sarthak Jain
  - name: Tannmay Yadav
  - name: Tessa D. Wilkinson

>   - name: Tiago Pereira

Reviewer 37: ```suggestion
  - name: Tiago M. D. Pereira
    orcid: 000-0003-4747-4329
    affiliation: 
    - "Rosseland Centre for Solar Physics, University of Oslo, P.O. Box 1029 Blindern, NO-0315 Oslo, Norway"
    - "Institute of Theoretical Astrophysics, University of Oslo, P.O. Box 1029 Blindern, NO-0315 Oslo, Norway"
```

  - name: Yudhik Agrawal
  - name: jamescalixto
  - name: yasintoda

> 

Reviewer 38: - name : Manas Mangaonkar

  - name: Emmanuel Arias
  - name: Fionnlagh Mackenzie Dover
    orcid: 0000-0002-1984-7303
    affiliation: "SP2RC, School of Mathematics and Statistics, The University of Sheffield, UK"
  - name: Freek Verstringe
    affiliation: "Royal Observatory of Belgium"
  - name: Gulshan Kumar
    orcid: 0000-0001-8523-7223
    affiliation: International Institute of Information Technology, Hyderabad
  - name: Harsh Mathur
    orcid: 0000-0001-5253-4213
    affiliation: Indian Institute of Astrophysics, Bangalore
  - name: Igor Babuschkin
  - name: Jaylen Wimbish
  - name: Juan Camilo Buitrago-Casas
  - name: Kalpesh Krishna
  - name: Kaustubh Hiware
    orcid: 0000-0003-3301-1016
    affiliation: "Indian Institute of Technology, Kharagpur"
  - name: Manas Mangaonkar
  - name: Matthew Mendero
  - name: Mickaël Schoentgen
  - name: Norbert Gyenge
  - name: Ole Streicher
    orcid: 0000-0001-7751-1843
    affiliation: Leibniz Institute for Astrophysics Potsdam, Germany
  - name: Rajasekhar Reddy Mekala
  - name: Rishabh Mishra
  - name: Shashank Srikanth
    affiliation: International Institute of Information Technology, Hyderabad
  - name: Sarthak Jain

>   - name: Tannmay Yadav

Author: Ported from a non-inline comment.
```suggestion
  - name: Tannmay Yadav
    orcid: 0000-0002-3143-5635
    affiliation: "Department of Chemical Engineering, Indian Institute of Technology Kharagpur, 721302, India"
```
```

  - Astronomy
  - Solar physics
authors:
  - name: Stuart J. Mumford
    orcid: 0000-0003-4217-4642
    affiliation: 
      - "SP2RC, School of Mathematics and Statistics, The University of Sheffield, UK"
      - "Aperio Software Ltd., Headingley Enterprise and Arts Centre, Bennett Road, Leeds LS6 3HN"
  - name: Nabil Freij
    orcid: 0000-0002-6253-082X
    affiliation: "Institute for Environmental Analytics, University of Reading, Reading RG6 6BX"
  - name: Steven Christe
    orcid: 0000-0001-6127-795X
    affiliation: "NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA"
  - name: Jack Ireland
  - name: Florian Mayer
  - name: Keith Hughitt
  - name: Albert Y. Shih
    orcid: 0000-0001-6874-2594
    affiliation: "NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA"
  - name: Daniel Ryan
  - name: Simon Liedtke
  - name: David Pérez-Suárez
    orcid: 0000-0003-0784-6909
    affiliation: "University College London, Gower Street, London, UK"
  - name: Pritish Chakraborty
  - name: Vishnunarayan K I.
  - name: Andrew Inglis
    orcid: 0000-0003-0656-2437
    affiliation: "Catholic University of America / NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA"

>   - name: Punyaslok Pattnaik

Reviewer 39: ```suggestion
  - name: Punyaslok Pattnaik
    orcid: 0000-0001-6216-2353
    affiliation: 'International Institute of Information Technology, Hyderabad 500 032, India'
```

  - name: Laura Hayes
    orcid: 0000-0002-6835-2390
    affiliation: 'NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA'
  - name: Asish Panda
  - name: Matt Earnshaw
  - name: Nitin Choudhary
    orcid : 0000-0001-6915-4583
    affiliation: 'Department of Mathematics, Indian Institute of Technology, Kharagpur, 721302, India'
  - name: Ankit Kumar
  - name: Prateek Chanda
    orcid: 0000-0002-7068-2866
    affiliation: 'Department of Computer Science & Technology, Indian Institute of Engineering Science & Technology, Shibpur, 711103, India'
  - name: Md Akramul Haque
    affiliation: 'Department of Mechanical Engineering, ZHCET, Aligarh Muslim University, Aligarh, India'
  - name: Michael S Kirk
    orcid: 0000-0001-9874-1429
    affiliation: 'Catholic University of America / NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA'
  - name: Michael Mueller
  - name: Sudarshan Konge
  - name: Rajul Srivastava
  - name: Yash Jain
  - name: Samuel Bennett
  - name: Ankit Baruah
  - name: Will Barnes
    orcid: 0000-0001-9642-6089
    affiliation: Lockheed Martin Solar and Astrophysics Laboratory / Bay Area Environmental Research Institute
  - name: Michael Charlton

>   - name: Cubostar

Author: ```suggestion
```

  - name: Shane Maloney
    orcid: 0000-0002-4715-1805
    affiliation: Trinity College Dublin / Dublin Institute for Advanced Studies
  - name: Nicky Chorley
    orcid: 0000-0002-2747-2716
    affiliation: Centre for Fusion, Space and Astrophysics, Physics Department, University of Warwick, Coventry CV4 7AL, United Kingdom
  - name: Himanshu
  - name: Sanskar Modi
  - name: James Paul Mason
  - name: Naman9639

>   - name: Yash

Author: ```suggestion
```

  - name: Ruben De Visscher
  - name: Shresth Verma
    orcid:  0000-0003-0370-5471
    affiliation: "ABV-Indian Institute of Information Technology and Management, Gwalior, MP 474015, India"
  - name: Ankit Agrawal
  - name: Dumindu Buddhika
  - name: Swapnil Sharma
  - name: Jongyeob Park
  - name: Matt Bates
  - name: Dhruv Goel
  - name: Garrison Taylor
  - name: Goran Cetusic
  - name: Jacob
  - name: Mateo Inchaurrandieta
  - name: Sally Dacie
  - name: Sanjeev Dubey
  - name: Deepankar Sharma
  - name: Erik M. Bray
  - name: Jai Ram Rideout
    orcid: 0000-0003-2587-1454
    affiliation: "Dogfox Software LLC, Flagstaff, AZ 86001, USA"
  - name: Serge Zahniy
  - name: Tomas Meszaros
  - name: Abhigyan Bose
  - name: Andre Chicrala
  - name: Ankit
  - name: Chloé Guennou
  - name: Daniel D'Avella
  - name: Daniel Williams
    affiliation: "SUPA, University of Glasgow, Glasgow G12 8QQ, United Kingdom"
    orcid: 0000-0003-3772-198X
  - name: Jordan Ballew
  - name: Nick Murphy
  - name: Priyank Lodha
  - name: Thomas Robitaille
  - name: Yash Krishan
  - name: Andrew Hill
  - name: Arthur Eigenbrot
    orcid: 0000-0003-0810-4368
    affiliation: "National Solar Observatory" 

>   - name: Benjamin Mampaey

Reviewer 40: ```suggestion
  - name: Benjamin Mampaey
    orcid: 0000-0001-6541-4966
    affiliation: Royal Observatory of Belgium
```

  - name: Freek Verstringe
    affiliation: "Royal Observatory of Belgium"
  - name: Gulshan Kumar
    orcid: 0000-0001-8523-7223
    affiliation: International Institute of Information Technology, Hyderabad
  - name: Harsh Mathur
    orcid: 0000-0001-5253-4213
    affiliation: Indian Institute of Astrophysics, Bangalore
  - name: Igor Babuschkin
  - name: Jaylen Wimbish
  - name: Juan Camilo Buitrago-Casas
  - name: Kalpesh Krishna
  - name: Kaustubh Hiware
    orcid: 0000-0003-3301-1016
    affiliation: "Indian Institute of Technology, Kharagpur"
  - name: Manas Mangaonkar
  - name: Matthew Mendero
  - name: Mickaël Schoentgen
  - name: Norbert Gyenge
  - name: Ole Streicher
    orcid: 0000-0001-7751-1843
    affiliation: Leibniz Institute for Astrophysics Potsdam, Germany
  - name: Rajasekhar Reddy Mekala
  - name: Rishabh Mishra
  - name: Shashank Srikanth
    affiliation: International Institute of Information Technology, Hyderabad
  - name: Sarthak Jain
  - name: Tannmay Yadav
    orcid: 0000-0002-3143-5635
    affiliation: "Department of Chemical Engineering, Indian Institute of Technology Kharagpur, 721302, India"
  - name: Tessa D. Wilkinson
  - name: Tiago M. D. Pereira

>     orcid: 000-0003-4747-4329

Author: ```suggestion
    orcid: 0000-0003-4747-4329
```

---
title: 'SunPy: A Python package for Solar Physics'
tags:
  - Python
  - Astronomy
  - Solar physics
authors:
  - name: Stuart J. Mumford
    orcid: 0000-0003-4217-4642
    affiliation: 
      - "SP2RC, School of Mathematics and Statistics, The University of Sheffield, UK"
      - "Aperio Software Ltd., Headingley Enterprise and Arts Centre, Bennett Road, Leeds LS6 3HN"
  - name: Nabil Freij
    orcid: 0000-0002-6253-082X
    affiliation: "Institute for Environmental Analytics, University of Reading, Reading RG6 6BX"
  - name: Steven Christe
    orcid: 0000-0001-6127-795X
    affiliation: "NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA"

>   - name: Jack Ireland

Reviewer 41: ```suggestion
  - name: Jack Ireland
     orcid: 0000-0002-2019-8881
     affiliation: "NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA"
```

  - name: Florian Mayer
  - name: Keith Hughitt
  - name: Albert Y. Shih
    orcid: 0000-0001-6874-2594
    affiliation: "NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA"
  - name: Daniel F. Ryan
    orcid: 0000-0001-8661-3825
    affiliation: 
      - "American University, Washington, DC 20016, USA"
      - "NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA"
  - name: Simon Liedtke
  - name: David Pérez-Suárez
    orcid: 0000-0003-0784-6909
    affiliation: "University College London, Gower Street, London, UK"

>   - name: Pritish Chakraborty

Reviewer 42: ```suggestion
  - name: Pritish Chakraborty
    orcid: 0000-0001-8875-5819
    affiliation: Manav Rachna University
```

  - name: David Stansby
    orcid: 0000-0002-1365-1908
    affiliation: 'Mullard Space Science Laboratory, University College London, Surrey, UK'
  - name: Russell Hewett
  - name: Alex Hamilton
  - name: Laura Hayes
    orcid: 0000-0002-6835-2390
    affiliation: 'NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA'
  - name: Asish Panda
  - name: Matt Earnshaw
  - name: Nitin Choudhary
    orcid : 0000-0001-6915-4583
    affiliation: 'Department of Mathematics, Indian Institute of Technology, Kharagpur, 721302, India'
  - name: Ankit Kumar
  - name: Prateek Chanda
    orcid: 0000-0002-7068-2866
    affiliation: 'Department of Computer Science & Technology, Indian Institute of Engineering Science & Technology, Shibpur, 711103, India'
  - name: Md Akramul Haque
    affiliation: 'Department of Mechanical Engineering, ZHCET, Aligarh Muslim University, Aligarh, India'
  - name: Michael S Kirk
    orcid: 0000-0001-9874-1429
    affiliation: 'Catholic University of America / NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA'
  - name: Michael Mueller
  - name: Sudarshan Konge
  - name: Rajul Srivastava

>   - name: Yash Jain

Reviewer 43: ```suggestion
   - name: Yash Jain
      orcid: 0000-0001-5347-4734
      affiliation: 'Indian Institute of Technology, Kharagpur, 721302, India'
```

  - name: Samuel Bennett
  - name: Ankit Baruah
  - name: Will Barnes
    orcid: 0000-0001-9642-6089
    affiliation: Lockheed Martin Solar and Astrophysics Laboratory / Bay Area Environmental Research Institute
  - name: Michael Charlton
  - name: Shane Maloney
    orcid: 0000-0002-4715-1805
    affiliation: Trinity College Dublin / Dublin Institute for Advanced Studies
  - name: Nicky Chorley
    orcid: 0000-0002-2747-2716
    affiliation: Centre for Fusion, Space and Astrophysics, Physics Department, University of Warwick, Coventry CV4 7AL, United Kingdom
  - name: Himanshu
  - name: Sanskar Modi

>   - name: James Paul Mason

Reviewer 44: name: James Paul Mason
orcid: 0000-0002-3783-5509
affiliation: NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA

  - name: Naman9639
  - name: Jose Ivan Campos Rozo
    orcid: 0000-0001-8883-6790
    affiliation: Institut für Physik/IGAM - Karl-Franzens University of Graz, Austria
  - name: Larry Manley

>   - name: Agneet Chatterjee

Author: ```suggestion
  - name: Agneet Chatterjee
    orcid : 0000-0002-0961-9569
    affiliation : Jadavpur University, Kolkata
```

  - name: John Evans
  - name: Michael Malocha
  - name: Monica G. Bobra
    affiliation: "W.W. Hansen Experimental Physics Laboratory, Stanford University, Stanford, CA 94305, USA"
    orcid: 0000-0002-5662-9604

>   - name: Sourav Ghosh

Author: ```suggestion
  - name: Sourav Ghosh
    orcid: 0000-0002-7259-5651
    affiliation: Jadavpur University, Kolkata
```

  - name: Airmansmith97
  - name: Dominik Stańczak
    affiliation: University of Warsaw
    orcid: 0000-0001-6291-8843
  - name: Ruben De Visscher
  - name: Shresth Verma
    orcid:  0000-0003-0370-5471
    affiliation: "ABV-Indian Institute of Information Technology and Management, Gwalior, MP 474015, India"
  - name: Ankit Agrawal
  - name: Dumindu Buddhika
  - name: Swapnil Sharma
  - name: Jongyeob Park
  - name: Matt Bates
  - name: Dhruv Goel
  - name: Garrison Taylor
  - name: Goran Cetusic
  - name: Jacob
  - name: Mateo Inchaurrandieta

>   - name: Sally Dacie

Reviewer 45: ```suggestion
  + name: Sally Dacie
  + orcid: 0000-0001-7572-2903
  + affiliation: Mullard Space Science Laboratory, University College London, Holmbury St. Mary, Surrey, RH5 6NT, UK
```

  - name: Sanjeev Dubey
  - name: Deepankar Sharma
  - name: Erik M. Bray
  - name: Jai Ram Rideout
    orcid: 0000-0003-2587-1454
    affiliation: "Dogfox Software LLC, Flagstaff, AZ 86001, USA"
  - name: Serge Zahniy
  - name: Tomas Meszaros
  - name: Abhigyan Bose

>   - name: Andre Chicrala

Author: ```suggestion
  - name: André Chicrala
    orcid: 0000-0002-5230-4909
    affiliation: Northumbria University, England
```

  - name: Ankit
  - name: Chloé Guennou
  - name: Daniel D'Avella
  - name: Daniel Williams
    affiliation: "SUPA, University of Glasgow, Glasgow G12 8QQ, United Kingdom"
    orcid: 0000-0003-3772-198X
  - name: Jordan Ballew

>   - name: Nick Murphy

Reviewer 46: ```suggestion
  - name: Nick Murphy
  - affiliation: Center for Astrophysics | Harvard & Smithsonian
  - orcid: 0000-0001-6628-8033
```

      - 'Department of Astronomy, University of Michigan, 1085 South University Avenue, Ann Arbor, MI 48109, USA'
  - name: Adrian Price-Whelan
    orcid: 0000-0003-0872-7098
    affiliation: Center for Computational Astrophysics, Flatiron Institute, 162 Fifth Ave, New York, NY 10010, USA
  - name: Ambar Mehrotra
  - name: Arseniy Kustov
  - name: Brandon Stone
  - name: Trung Kien Dang
    orcid: 0000-0001-7562-6495
    affiliation: Saw Swee Hock School of Public Health, National University Health System, National University of Singapore, Singapore
  - name: Emmanuel Arias
  - name: Fionnlagh Mackenzie Dover
    orcid: 0000-0002-1984-7303
    affiliation: "SP2RC, School of Mathematics and Statistics, The University of Sheffield, UK"
  - name: Freek Verstringe
    affiliation: "Royal Observatory of Belgium"
  - name: Gulshan Kumar
    orcid: 0000-0001-8523-7223
    affiliation: International Institute of Information Technology, Hyderabad
  - name: Harsh Mathur
    orcid: 0000-0001-5253-4213
    affiliation: Indian Institute of Astrophysics, Bangalore
  - name: Igor Babuschkin
  - name: Jaylen Wimbish
  - name: Juan Camilo Buitrago-Casas

>   - name: Kalpesh Krishna

Reviewer 47: ```suggestion
  - name: Kalpesh Krishna
    orcid: 0000-0001-6574-0817
    affiliation: University of Massachusetts, Amherst
```

  - name: Kaustubh Hiware
    orcid: 0000-0003-3301-1016
    affiliation: "Indian Institute of Technology, Kharagpur"
  - name: Manas Mangaonkar
  - name: Matthew Mendero

>   - name: Mickaël Schoentgen

Reviewer 48: ```suggestion
  - name: Mickaël Schoentgen
    orcid: 0000-0002-0106-4810
```

Reviewer 48: FTR I have no affiliation, and do not want to put my current employer as it is not related. Is it OK?

Author: That's no problem :smile:

  - name: Norbert Gyenge
  - name: Ole Streicher
    orcid: 0000-0001-7751-1843
    affiliation: Leibniz Institute for Astrophysics Potsdam, Germany
  - name: Rajasekhar Reddy Mekala
  - name: Rishabh Mishra
  - name: Shashank Srikanth
    affiliation: International Institute of Information Technology, Hyderabad
  - name: Sarthak Jain
  - name: Tannmay Yadav
    orcid: 0000-0002-3143-5635
    affiliation: "Department of Chemical Engineering, Indian Institute of Technology Kharagpur, 721302, India"
  - name: Tessa D. Wilkinson
  - name: Tiago M. D. Pereira
    orcid: 0000-0003-4747-4329
    affiliation: 
      - "Rosseland Centre for Solar Physics, University of Oslo, P.O. Box 1029 Blindern, NO-0315 Oslo, Norway"
      - "Institute of Theoretical Astrophysics, University of Oslo, P.O. Box 1029 Blindern, NO-0315 Oslo, Norway"
  - name: Yudhik Agrawal
  - name: jamescalixto
  - name: yasintoda

> 

Reviewer 49: - name: Sophie A. Murray
 orcid: 0000-0002-9378-5315
 affiliation: Trinity College Dublin, Ireland

---
title: 'SunPy: A Python package for Solar Physics'
tags:
  - Python
  - Astronomy
  - Solar physics
authors:
  - name: Stuart J. Mumford
    orcid: 0000-0003-4217-4642
    affiliation: 
      - "SP2RC, School of Mathematics and Statistics, The University of Sheffield, UK"
      - "Aperio Software Ltd., Headingley Enterprise and Arts Centre, Bennett Road, Leeds LS6 3HN"
  - name: Nabil Freij
    orcid: 0000-0002-6253-082X
    affiliation: "Institute for Environmental Analytics, University of Reading, Reading RG6 6BX"
  - name: Steven Christe
    orcid: 0000-0001-6127-795X
    affiliation: "NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA"
  - name: Jack Ireland
    orcid: 0000-0002-2019-8881
    affiliation: "NASA Goddard Space Flight Center, Greenbelt, MD 20771, USA"
  - name: Florian Mayer

>   - name: Keith Hughitt

Reviewer 50: ```suggestion
  - name: V. Keith Hughitt
    orcid: 0000-0003-0787-9559
    affiliation: "Center for Cancer Research, National Cancer Institute, Bethesda, MD 20892-9760, USA"
```

Author: :wave: Hi Keith!

The `sunpy` Python package is a community-developed, free, and open-source solar data analysis environment for Python that provides a comprehensive set of tools for performing common tasks in solar data analysis. It is managed by the SunPy Project, an organization that facilitates and promotes the use of open development and open source packages like `sunpy` through community engagement and tools such as [GitHub](https://github.com/sunpy/sunpy/), [mailing lists](https://groups.google.com/forum/#!forum/sunpy), and [matrix](https://matrix.to/#/#sunpy:openastronomy.org).

The four most significant subpackages of `sunpy` are described below.

The `sunpy.net` subpackage provides a unified interface that simplifies and homogenizes search and retrieval by querying and downloading data from many solar data sources, irrespective of the underlying data-source client. It currently supports sourcing data from 18 different space- and ground-based solar observatories.

> The `sunpy.map` and `sunpy.timeseries` subpackages provide core data types (`Map` and `TimeSeries`, respectively) that are designed to provide a general, standard, and consistent interface for loading and representing solar data across different instruments and missions. These classes load data which conform to solar physics standards and conventions such as FITS [@wells_fits_1981], FITS World Coordinate Systems (WCS) [@fits_wcs], and solar-specific FITS headers [@thompson_coordinates_2006], while allowing customization to account for differences in specific instruments. Visualization methods are also provided to inspect and plot those data. Example visulizations of both `TimeSeries` and `Map` are shown in Figure 1.

Reviewer 51: The reference `@thompson_coordinates_2006` is showing up in [this rendered PDF](https://github.com/openjournals/joss-reviews/files/3829309/10.21105.joss.01832.pdf) as `???`. Looks like it is `coordinate` (singular) in the BiBTeX entry, not plural.

Reviewer 19: 👍 Fixed by https://github.com/sunpy/sunpy/commit/6188dad4528161f7096954cd569f6d9399d8b697

![Left: An example of `TimeSeries` for the GOES X-ray Sensor in two broadband channels. Right: A `Map` of the extreme ultraviolet 171 $\mathring A$ channel of AIA corresponding to the time of a solar flare depicted by the vertical dashed line in the left-hand panel.](https://codimd.s3.shivering-isles.com/demo/uploads/upload_5eabd346e54c29b1d9b74aa11e351b30.png)

The `sunpy.coordinates` subpackage provides support for representing and transforming coordinates used in solar physics and astrophysics. These coordinates may represent events (e.g., flares), features on or above the Sun (e.g., magnetic loops), or the position of structures traveling throughout the heliosphere (e.g., coronal mass ejections). The package currently implements the most widely used Sun-centered coordinate frames, and extends `astropy.coordinates`.

Other functionality provided by `sunpy` includes physical models of solar behavior, such as differential rotation, color maps for certain data sources, image-processing routines integrated with `Map`, and useful physical parameters such as constants.

> The `sunpy` package is designed to be extensible, which means that it is easy to add support for additional instruments or data sources. It relies heavily on the `astropy` Python package as well as the scientific python stack (e.g. `numpy`, `scipy`, `matplotlib` and `pandas`).

Reviewer 51: You should probably cite `astropy` and consider citing the others you list.

Reviewer 19: 👍 Fixed by https://github.com/sunpy/sunpy/commit/1c4dc21bf044580822db5cbffa46a30578663163

A more complete description of the SunPy Project and the `sunpy` package, the methodology, development model, and implementation can be found in [@sunpy_community2019].

> The SunPy Project supports affiliated packages, which build upon or extends the functionality of `sunpy`. Current affiliated packages are `drms` [@Glogowski2019drms], [`ndcube`](https://docs.sunpy.org/projects/ndcube), [`radiospectra`](https://docs.sunpy.org/projects/radiospectra) and [`IRISPy`](https://docs.sunpy.org/projects/irispy). The Project is also a member of the Python in Heliophysics community [PyHC, @annex], whose mission is to enable interdisciplinary analysis across all sub-disciplines of heliophysics by adhering to standards for code development and interoperability.

Reviewer 51: You should add a link to `drms` like you have with the other affiliated packages.

Reviewer 19: 👍 Fixed by https://github.com/sunpy/sunpy/commit/6188dad4528161f7096954cd569f6d9399d8b697

![Left: An example of `TimeSeries` for the GOES X-ray Sensor in two broadband channels. Right: A `Map` of the extreme ultraviolet 171 $\mathring A$ channel of AIA corresponding to the time of a solar flare depicted by the vertical dashed line in the left-hand panel.](https://codimd.s3.shivering-isles.com/demo/uploads/upload_5eabd346e54c29b1d9b74aa11e351b30.png)

The `sunpy.coordinates` subpackage provides support for representing and transforming coordinates used in solar physics and astrophysics. These coordinates may represent events (e.g., flares), features on or above the Sun (e.g., magnetic loops), or the position of structures traveling throughout the heliosphere (e.g., coronal mass ejections). The package currently implements the most widely used Sun-centered coordinate frames, and extends `astropy.coordinates`.

Other functionality provided by `sunpy` includes physical models of solar behavior, such as differential rotation, color maps for certain data sources, image-processing routines integrated with `Map`, and useful physical parameters such as constants.

> The `sunpy` package is designed to be extensible, which means that it is easy to add support for additional instruments or data sources. It relies heavily on the `astropy`[@astropy] Python package as well as the scientific python stack (e.g. `numpy`[@numpy], `scipy`[@scipy], `matplotlib`[@matplotlib] and `pandas`[@pandas]).

Reviewer 51: ```suggestion
The `sunpy` package is designed to be extensible, which means that it is easy to add support for additional instruments or data sources. It relies heavily on the `astropy` [@astropy] Python package as well as the scientific python stack (e.g. `numpy` [@numpy], `scipy` [@scipy], `matplotlib` [@matplotlib] and `pandas` [@pandas]).
```

The rendered version of the paper looks a little funny without spaces between the package names and the citations (see [this version](https://github.com/openjournals/joss-papers/blob/joss.01832/joss.01832/10.21105.joss.01832.pdf)) so suggest adding those here.

  index: 47
- name: Leibniz Institute for Astrophysics Potsdam, Germany
  index: 48
- name: Department of Chemical Engineering, Indian Institute of Technology Kharagpur, 721302, India
  index: 49
- name: Rosseland Centre for Solar Physics, University of Oslo, P.O. Box 1029 Blindern, NO-0315 Oslo, Norway
  index: 50
- name: Institute of Theoretical Astrophysics, University of Oslo, P.O. Box 1029 Blindern, NO-0315 Oslo, Norway
  index: 51
- name: Trinity College Dublin, Ireland
  index: 52
date: 3 October 2019
bibliography: paper.bib
aas-doi: 10.3847/1538-4357/ab4f7a
aas-journal: The Astrophysical Journal
---

# Summary

The Sun, our nearest star, is a local laboratory for studying universal physical processes. Solar physics as a discipline includes studying the Sun both as a star and as the primary driver of space weather throughout the heliosphere. Due to the Sun's proximity, the temporal and spatial resolution of solar observations are orders of magnitude larger than those of other stars. This leads to significant differences in the data-analysis software needs of solar physicists compared with astrophysicists.

> The `sunpy` Python package is a community-developed, free, and open-source solar data analysis environment for Python that provides a comprehensive set of tools for performing common tasks in solar data analysis. It is managed by the SunPy Project, an organization that facilitates and promotes the use of open development and open source packages like `sunpy` through community engagement and tools such as [GitHub](https://github.com/sunpy/sunpy/), [mailing lists](https://groups.google.com/forum/#!forum/sunpy), and [matrix](https://matrix.to/#/#sunpy:openastronomy.org).

Reviewer 14: Removed the end of the first sentence (starting with "that provides...") as it is redundant. 
```suggestion
The `sunpy` Python package is a community-developed, free, and open-source solar data analysis environment for Python. It is managed by the SunPy Project, an organization that facilitates and promotes the use of open development and open source packages like `sunpy` through community engagement and tools such as [GitHub](https://github.com/sunpy/sunpy/), [mailing lists](https://groups.google.com/forum/#!forum/sunpy), and [matrix](https://matrix.to/#/#sunpy:openastronomy.org).
```

Other functionality provided by `sunpy` includes physical models of solar behavior, such as differential rotation, color maps for certain data sources, image-processing routines integrated with `Map`, and useful physical parameters such as constants.

The `sunpy` package is designed to be extensible, which means that it is easy to add support for additional instruments or data sources. It relies heavily on the `astropy` [@astropy] Python package as well as the scientific python stack (e.g. `numpy` [@numpy], `scipy` [@scipy], `matplotlib` [@matplotlib] and `pandas` [@pandas]).

A more complete description of the SunPy Project and the `sunpy` package, including the methodology, development model, and implementation, as well as a comparison with other commonly-used packages in solar physics, can be found in [@sunpy_community2019].

> The SunPy Project supports affiliated packages, which build upon or extends the functionality of `sunpy`. Current affiliated packages are [`drms`](https://docs.sunpy.org/projects/drms) [@Glogowski2019drms], [`ndcube`](https://docs.sunpy.org/projects/ndcube), [`radiospectra`](https://docs.sunpy.org/projects/radiospectra) and [`IRISPy`](https://docs.sunpy.org/projects/irispy). The Project is also a member of the Python in Heliophysics community [PyHC, @annex], whose mission is to enable interdisciplinary analysis across all sub-disciplines of heliophysics by adhering to standards for code development and interoperability.

Reviewer 14: ```suggestion
The SunPy Project supports affiliated packages, which build upon or extend the functionality of `sunpy`. The current affiliated packages are [`drms`](https://docs.sunpy.org/projects/drms) [@Glogowski2019drms], [`ndcube`](https://docs.sunpy.org/projects/ndcube), [`radiospectra`](https://docs.sunpy.org/projects/radiospectra) and [`sunraster`](https://docs.sunpy.org/projects/sunraster). The Project is also a member of the Python in Heliophysics community [PyHC, @annex], whose mission is to enable interdisciplinary analysis across all sub-disciplines of heliophysics by adhering to standards for code development and interoperability.
```
